cmd/backup.go

```diff
@@ -0,0 +1,141 @@
+package cmd
+
+import (
+	"context"
+
+	"github.com/AlecAivazis/survey/v2"
+	pb "github.com/odpf/optimus/api/proto/odpf/optimus"
+	"github.com/odpf/optimus/config"
+	"github.com/odpf/optimus/models"
+	"github.com/odpf/salt/log"
+	"github.com/pkg/errors"
+	cli "github.com/spf13/cobra"
+	"google.golang.org/grpc"
+)
+
+func backupCommand(l log.Logger, datastoreRepo models.DatastoreRepo, conf config.Provider) *cli.Command {
+	cmd := &cli.Command{
+		Use:   "backup",
+		Short: "Backup a resource and its downstream",
+		Long:  "Backup supported resource of a datastore and all of its downstream dependencies",
+	}
+	cmd.AddCommand(backupResourceSubCommand(l, datastoreRepo, conf))
+	return cmd
+}
+
+func backupResourceSubCommand(l log.Logger, datastoreRepo models.DatastoreRepo, conf config.Provider) *cli.Command {
+	backupCmd := &cli.Command{
+		Use:   "resource",
+		Short: "backup a resource",
+	}
+
+	dryRun := false
+	var (
+		project   string
+		namespace string
+	)
+
+	backupCmd.Flags().BoolVarP(&dryRun, "dry-run", "", dryRun, "do a trial run with no permanent changes")
+	backupCmd.Flags().StringVarP(&project, "project", "p", "", "project name of optimus managed ocean repository")
+	backupCmd.MarkFlagRequired("project")
+	backupCmd.Flags().StringVarP(&namespace, "namespace", "n", "", "namespace of the requester")
+	backupCmd.MarkFlagRequired("namespace")
+
+	backupCmd.RunE = func(cmd *cli.Command, args []string) error {
+		availableStorer := []string{}
+		for _, s := range datastoreRepo.GetAll() {
+			availableStorer = append(availableStorer, s.Name())
+		}
+		var storerName string
+		if err := survey.AskOne(&survey.Select{
+			Message: "Select supported datastores?",
+			Options: availableStorer,
+		}, &storerName); err != nil {
+			return err
+		}
+
+		var qs = []*survey.Question{
+			{
+				Name: "name",
+				Prompt: &survey.Input{
+					Message: "What is the resource name?",
+					Help:    "Input urn of the resource",
+				},
+				Validate: survey.ComposeValidators(validateNoSlash, survey.MinLength(3),
+					survey.MaxLength(1024)),
+			},
+			{
+				Name: "description",
+				Prompt: &survey.Input{
+					Message: "Why is this backup needed?",
+					Help:    "Describe backup intention to help you identify backup",
+				},
+				Validate: survey.ComposeValidators(validateNoSlash, survey.MinLength(3),
+					survey.MaxLength(1024)),
+			},
+			{
+				Name: "backupDownstream",
+				Prompt: &survey.Confirm{
+					Message: "Backup downstream?",
+					Help:    "Select yes to also backup the downstream resources",
+				},
+			},
+		}
+		inputs := map[string]interface{}{}
+		if err := survey.Ask(qs, &inputs); err != nil {
+			return err
+		}
+		resourceName := inputs["name"].(string)
+		description := inputs["description"].(string)
+		backupDownstream := inputs["backupDownstream"].(bool)
+		backupRequest := &pb.BackupDryRunRequest{
```

Comment:
does check means that if dry run, we will use `pb.BackupDryRunRequest` and if not, use `pb.BackupRequest`? 

my plan for this backup flow is:
1. do backup dry run ---> this will return list of resources to be backed up
2. if user specify the dry run option, it will end here. if not, it will ask user confirmation to proceed running the backup
3. do the backup

based on the flow,  whether the user chooses to do dry run or not, it will still do dry run, that is why there is no check for it. the check will be done before running the actual backup.

what do you think?